### PR TITLE
feat(webresources): pull and push CLI commands with hash tracking

### LINKED ISSUE #161: feat(webresources): Add pull command with hash tracking
## Summary

Add `ppds webresources pull` command to download multiple resources with local tracking for conflict detection.

## Depends On

- #160 (list and get commands)

## Command

```bash
ppds webresources pull <folder>
                       [--solution <name>]
                       [--type <type>]
                       [--name <pattern>]
                       [--strip-prefix]       # Remove publisher prefix from paths
                       [--force]              # Overwrite existing without prompt
```

## Tasks

- [ ] Implement `PullCommand.cs`
- [ ] Create `.ppds/webresources.json` tracking file in target folder
- [ ] Create hierarchical directory structure preserving resource paths
- [ ] `--strip-prefix` option to remove publisher prefix (e.g., `new_/`)
- [ ] `--force` to overwrite existing files
- [ ] Track both `modifiedOn` timestamp and SHA256 hash per resource
- [ ] Download content in parallel for performance
- [ ] Unit tests

## Tracking File Schema

`.ppds/webresources.json`:
```json
{
  "version": 1,
  "environment": "https://org.crm.dynamics.com",
  "resources": {
    "new_/scripts/myfile.js": {
      "id": "00000000-0000-0000-0000-000000000000",
      "modifiedOn": "2026-01-04T12:00:00Z",
      "hash": "sha256:abc123...",
      "pulledAt": "2026-01-04T12:00:00Z",
      "localPath": "scripts/myfile.js"
    }
  }
}
```

## Directory Structure

**Default (hierarchical):**
```
output/
├── .ppds/
│   └── webresources.json
├── new_/
│   └── scripts/
│       └── myfile.js
└── new_/
    └── styles/
        └── main.css
```

**With `--strip-prefix`:**
```
output/
├── .ppds/
│   └── webresources.json
├── scripts/
│   └── myfile.js
└── styles/
    └── main.css
```

## Files to Create

```
src/PPDS.Cli/Commands/WebResources/
├── PullCommand.cs
└── Models/
    └── WebResourceTrackingFile.cs

tests/PPDS.Cli.Tests/Commands/WebResources/
└── PullCommandTests.cs
```

## References

- Parent issue: #141

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### LINKED ISSUE #162: feat(webresources): Add push command with conflict detection
## Summary

Add `ppds webresources push` command to upload web resources with conflict detection based on server timestamps.

## Depends On

- #161 (pull command with tracking)

## Command

```bash
ppds webresources push <path>                 # File or folder
                       [--solution <name>]    # Add to solution if not present
                       [--force]              # Skip conflict check
                       [--dry-run]            # Preview only (ADR-0013)
                       [--publish]            # Publish after push
```

## Behavior

1. Read `.ppds/webresources.json` from folder (or parent folders)
2. For each local file:
   - Query server `modifiedOn` (single-field query, lightweight)
   - Compare with tracked `modifiedOn` from pull
   - If different: **conflict detected**
3. If conflicts and no `--force`: exit code 1 with message
4. If `--force` or no conflicts: upload content
5. If `--publish`: publish after successful upload

## Conflict Detection Flow

```
┌─────────────────┐
│ Read local file │
└────────┬────────┘
         │
         ▼
┌────────────────────┐
│ Find in tracking   │
│ file by name       │
└────────┬───────────┘
         │
         ▼
┌────────────────────────────┐
│ Query server modifiedOn    │
│ (single field, fast)       │
└────────┬───────────────────┘
         │
         ▼
┌────────────────────────────┐
│ modifiedOn matches?        │
├───────────┬────────────────┤
│   Yes     │       No       │
│   ▼       │       ▼        │
│ Upload    │  CONFLICT      │
│           │  (exit 1 or    │
│           │   --force)     │
└───────────┴────────────────┘
```

## Tasks

- [ ] Implement `PushCommand.cs`
- [ ] Read tracking file to get expected `modifiedOn` values
- [ ] Query server for current `modifiedOn` (lightweight check)
- [ ] Conflict detection with clear error message
- [ ] `--force` to skip conflict check
- [ ] `--dry-run` per ADR-0013 (preview changes)
- [ ] `--publish` to publish after push
- [ ] `--solution` to add resource to solution if not already present
- [ ] Update tracking file after successful push
- [ ] Exit code 1 for conflicts, 0 for success
- [ ] Unit tests

## Error Output (Conflict)

```
Conflict detected: new_/scripts/myfile.js
  Local modified:  2026-01-04T10:00:00Z
  Server modified: 2026-01-04T11:30:00Z

Server was modified since last pull. Use one of:
  ppds webresources diff new_/scripts/myfile.js  # View changes
  ppds webresources push --force                  # Overwrite server
  ppds webresources pull --force                  # Discard local
```

## Files to Create

```
src/PPDS.Cli/Commands/WebResources/
└── PushCommand.cs

tests/PPDS.Cli.Tests/Commands/WebResources/
└── PushCommandTests.cs
```

## References

- Parent issue: #141
- [ADR-0013](docs/adr/0013_CLI_DRY_RUN_CONVENTION.md) - `--dry-run` convention
- Extension conflict detection: `WebResourceFileSystemProvider.ts:689-748`

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary
- Adds `ppds webresources pull <folder>` and `ppds webresources push <folder>` for round-trip web-resource editing with `.ppds/webresources.json` tracking.
- Pull supports `--solution`, `--type`, `--name`, `--strip-prefix`, `--force`; push supports `--force`, `--dry-run`, `--publish` with SHA-256 hash + `modifiedOn` conflict detection (exit 10 = `PreconditionFailed` on server-side drift).
- All business logic lives in `IWebResourceSyncService` (Constitution A1); commands are thin wrappers. Pull rejects path-traversal names and strip-prefix path collisions instead of silently overwriting.

[size-waived: cohesive feature spanning spec + tracking model + sync service + two CLI commands + tests for issues #161/#162; splitting would create non-functional intermediate PRs since the commands depend on the sync service which depends on the tracking model]

Closes #161
Closes #162

## Test Plan
- [x] `dotnet test PPDS.sln --filter "Category!=Integration"` — 3550 PPDS.Cli tests pass across net8/9/10
- [x] AC-WR-34 through AC-WR-55 all green; spec rows reference real test methods
- [x] Round-trip pull → edit → push --dry-run against live `PPDS Demo - Dev` env (1 of 20686 resources, `--name ppds_`)
- [x] Push with stale tracking exits 10 and prints tracked-vs-server timestamps
- [x] Push with missing tracking errors with `Run 'ppds webresources pull <folder>' first`
- [x] Strip-prefix collision regression test (`PullRejectsStripPrefixCollisions`) added

## Verification
- [x] /gates passed
- [x] /verify cli completed (round-trip + edge cases on live env)
- [x] /qa cli completed (10 findings, 6 fixed)
- [x] /review completed (26 findings, addressed in converge round 1)
- [x] Pre-PR self-review (round 2) found 4 DEFECTs — all fixed in commit 9d9f36f01

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/PPDS.Cli/Services/WebResources/WebResourceSyncService.cs [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+/// </summary>
+public class WebResourceSyncService : IWebResourceSyncService
+{
+    private const int DefaultDownloadParallelism = 8;
+    private const int DefaultUploadParallelism = 4;
+
+    // Path comparisons must be case-sensitive on POSIX, case-insensitive on Windows.
+    private static readonly StringComparison PathComparison = OperatingSystem.IsWindows()
+        ? StringComparison.OrdinalIgnoreCase
+        : StringComparison.Ordinal;
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Services/WebResources/WebResourceSyncService.cs]: Fixed in 009706ddb669e1e75469575ec15f3ecee1ec5c95 — Updated PathComparison to treat macOS as case-insensitive (default APFS/HFS+ are case-insensitive). Introduced PathsAreCaseInsensitive flag and shared PathComparer to avoid drift between this and the related pathClaims fix.

File: src/PPDS.Cli/Services/WebResources/WebResourceSyncService.cs [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+
+        var pulled = new List<PulledResource>();
+        var skipped = new List<SkippedResource>();
+        var errors = new List<ErrorResource>();
+        var newResourceEntries = new Dictionary<string, TrackedResource>(StringComparer.OrdinalIgnoreCase);
+
+        // Per-resource path resolution + traversal validation up front so we can
+        // skip downloads for invalid entries without occupying a download slot.
+        var processable = new List<(WebResourceInfo Resource, string LocalPath, string AbsolutePath)>();
+        var pathClaims = new Dictionary<string, string>(OperatingSystem.IsWindows() ? StringComparer.OrdinalIgnoreCase : StringComparer.Ordinal);
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Services/WebResources/WebResourceSyncService.cs]: Fixed in 009706ddb669e1e75469575ec15f3ecee1ec5c95 — pathClaims dictionary now uses the shared PathComparer, so local path collision detection is case-insensitive on macOS as well as Windows.
